### PR TITLE
Run CI on Windows

### PR DESCRIPTION
**Pull request by ark0f**

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -326,7 +326,47 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           path: artifact
+  
+  build-win:
+    needs: spec_version
+    runs-on: windows-latest
```
Comment (shamilsan) [inline .github/workflows/CI.yaml:332]: I guess `windows-latest` host will be too slow. Nightly build is about 50 mins: https://github.com/gear-tech/builds/runs/7014779874?check_suite_focus=true

Comment (ark0f) [reply .github/workflows/CI.yaml:332]: We build in debug mode so I expect it will be faster

**Comment by ark0f** [comment]
> Do we need to add a cache stage?

Of course, we do. But we test build time on GH runner for now

**Comment by shamilsan** [comment]
@sergeyfilyanin Please configure a Windows self-hosted server. Feel free to commit to this PR branch.

**Comment by ukint-vs** [comment]
> @ukint-vs Please consider integrating `cachepot` to this CI too. If it is quite a long task we may implement it as a separate PR.

I can do it in this PR

**Review by ukint-vs — APPROVED** [review]

**Pull request merged**